Metalama.Framework.Engine/Linking/LinkerLateTransformationRegistry.cs

```diff
@@ -0,0 +1,145 @@
+                                    case PropertyDeclarationSyntax propertyDeclaration:
+                                        if ( propertyDeclaration.Initializer == null )
+                                        {
+                                            continue;
+                                        }
+
+                                        primaryConstructorInitializedMembers.Add( propertySymbol );
+
+                                        break;
+
+                                    case ParameterSyntax:
+                                        primaryConstructorInitializedMembers.Add( propertySymbol );
+
+                                        break;
+                                }
+
+                                break;
+
+                            case IEventSymbol eventSymbol:
+                                var eventDeclaration = eventSymbol.GetPrimaryDeclaration().AssertNotNull();
+
+                                if ( eventDeclaration is VariableDeclaratorSyntax eventFieldDeclarator )
+                                {
+
+                                    if ( eventFieldDeclarator.Initializer == null )
+                                    {
+                                        continue;
+                                    }
+
+                                    primaryConstructorInitializedMembers.Add( eventSymbol );
+                                }
+
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+
+        public bool HasRemovedPrimaryConstructor(INamedTypeSymbol type)
+        {
+            return this._typesWithRemovedPrimaryConstructor.Contains(type);
+        }
+
+#pragma warning disable CA1822 // Mark members as static
+        public IReadOnlyList<IFieldSymbol> GetPrimaryConstructorFields( INamedTypeSymbol type )
+#pragma warning restore CA1822 // Mark members as static
+        {
+            return type.GetMembers().OfType<IFieldSymbol>().Where(f => f.Name is [ '<', .., '>', 'P' ] ).ToArray();
+        }
+
+#pragma warning disable CA1822 // Mark members as static
+        public IReadOnlyList<IPropertySymbol> GetPrimaryConstructorProperties( INamedTypeSymbol type )
+#pragma warning restore CA1822 // Mark members as static
+        {
+            return type.GetMembers().OfType<IPropertySymbol>().Where( p => p.GetPrimaryDeclaration() is ParameterSyntax ).ToArray();
+        }
+
+        public bool IsPrimaryConstructorInitializedMember( ISymbol symbol )
+        {
+            return this._primaryConstructorInitializedMembers.Contains( symbol );
+        }
+
+        public ArgumentListSyntax? GetPrimaryConstructorBaseArgumentList( IMethodSymbol constructor )
+        {
+            var type = constructor.ContainingType;
+
+            Invariant.Assert( this.HasRemovedPrimaryConstructor( type ) );
+
+            var typeSyntax = (TypeDeclarationSyntax) type.GetPrimaryDeclaration().AssertNotNull();
```

Comment:
Only one declaration of a `partial` type can have primary constructor, so it would be more correct to choose that one here and not the "primary" one.

Though I don't know if we care about that case.